python/fusion_engine_client/analysis/file_reader.py

```diff
@@ -302,336 +300,177 @@ def read(self, message_types: Union[list, tuple, set] = None,
-                        self.logger.debug('  Message before requested time range. Discarding. [time=%s (%s)]' %
-                                          (str(message_time_sec), time_type))
-                        continue
-                    elif time_range[1] is not None and time_offset_sec > float(time_range[1]):
-                        # Assuming data is in order so if we pass the end of the time range we're done.
-                        if generate_index:
-                            self.logger.debug('  Message past requested time range. Discarding. [time=%s (%s)]' %
-                                              (str(message_time_sec), time_type))
-                            continue
-                        else:
-                            self.logger.debug('  Message past requested time range. Done reading. [time=%s (%s)]' %
-                                              (str(message_time_sec), time_type))
-                            break
-
-                # Store the message.
-                #
-                # If we haven't reached the max message count yet, or it is 0 (disabled), store the message by type. If
-                # max_messages is negative, the user requested the N latest messages. Add them to an N-length circular
-                # buffer and pick off the newest ones when we're done with the file.
-                message_count += 1
-                if max_messages < 0:
-                    newest_messages.append((header, contents))
-                    bytes_used += message_size_bytes
-                elif max_messages == 0 or message_count <= abs(max_messages):
-                    self.data[header.message_type].messages.append(contents)
-                    bytes_used += message_size_bytes
-
-                if max_messages > 0:
-                    if generate_index and message_count > max_messages:
-                        self.logger.debug('  Max messages reached. Discarding. [# messages=%d]' % message_count)
-                        continue
-                    elif not generate_index and message_count == max_messages:
-                        self.logger.debug('  Max messages reached. Done reading. [# messages=%d]' % message_count)
-                        break
-            except Exception as e:
-                # Since the CRC passed we know we read the correct number of bytes, so if the decode fails because of
-                # a format version mismatch or something, we can simply skip this message. No need to stop reading the
-                # file.
-                self.logger.warning('Error decoding %s payload @ %d: %s' %
-                                    (header.get_type_string(), message_offset_bytes, repr(e)))
+        # Fast-forward the reader to EOF to print out one last progress update. If we already reached EOF and printed
+        # the update, this should have no effect.
+        self.reader.seek_to_eof()
```

Comment:
Maybe I'm missing something, but is the logic around forcing the EOF just to print out a progress message? It seems like a non-trivial amount of complexity for a log message that doesn't add much context.

Reply:
Yeah more or less. While I was debugging I found having that last printout was actually pretty useful, especially since we do a lot of reads in sequence in `p1_display`. I suppose I could change this to be a bit more direct and have an explicit print last call instead of actually seeking to eof. The benefit with doing it this way though, is there's no risk of printing 100% and then reading some more data.